src/transpile.rs

```diff
@@ -985,6 +1049,10 @@ impl ConnectionBuilder {
         let limit = self.limit_clause();
         let offset = self.offset.unwrap_or(0);
 
+        // Determine if aggregates are requested based on if we generated a select list
+        let requested_aggregates =
+            self.aggregate_builder.is_some() && aggregate_select_list.is_some();
```

Comment:
Don't need to check `self.aggregate_builder.is_some()` here because the `aggregate_select_list` function returns `None` if `self.aggregate_builder` is `None`.

Reply:
Fixed